ivadomed/loader/bids_dataframe.py

```diff
@@ -0,0 +1,298 @@
+import copy
+import itertools
+import os
+
+import bids as pybids
+import pandas as pd
+from loguru import logger
+
+
+class BidsDataframe:
+    """
+    This class aims to create a dataframe containing all BIDS image files in a list of path_data and their metadata.
+
+    Args:
+        loader_params (dict): Loader parameters, see :doc:`configuration_file` for more details.
+        path_output (str): Output folder.
+        derivatives (bool): If True, derivatives are indexed.
+
+    Attributes:
+        path_data (list): Paths to the BIDS datasets.
+        bids_config (str): Path to the custom BIDS configuration file.
+        target_suffix (list of str): List of suffix of targetted structures.
+        roi_suffix (str): List of suffix of ROI masks.
+        extensions (list of str): List of file extensions of interest.
+        contrast_lst (list of str): List of the contrasts of interest.
+        derivatives (bool): If True, derivatives are indexed.
+        df (pd.DataFrame): Dataframe containing dataset information
+    """
+
+    def __init__(self, loader_params, path_output, derivatives):
+
+        # paths_data from loader parameters
+        self.paths_data = loader_params['path_data']
+
+        # bids_config from loader parameters
+        self.bids_config = None if 'bids_config' not in loader_params else loader_params['bids_config']
+
+        # target_suffix and roi_suffix from loader parameters
+        self.target_suffix = copy.deepcopy(loader_params['target_suffix'])
+        # If `target_suffix` is a list of lists convert to list
```

Comment:
This is me being nit picky but it would be nice to have a line space between the previous line `self.target_suffix` and the next line with comments for code clarity.

Reply:
Yup. Great point! :)